metaflow/plugins/kfp/kfp.py

```diff
@@ -453,24 +467,28 @@ def _step_cli(self, node: DAGNode, task_id: str, user_code_retries: int) -> str:
 
     @staticmethod
     def _set_container_settings(container_op: ContainerOp, kfp_component: KfpComponent):
-        resource_requirements: Dict[str, str] = kfp_component.resource_requirements
-        if "memory" in resource_requirements:
-            container_op.container.set_memory_request(resource_requirements["memory"])
-        if "memory_limit" in resource_requirements:
-            container_op.container.set_memory_limit(
-                resource_requirements["memory_limit"]
-            )
-        if "cpu" in resource_requirements:
-            container_op.container.set_cpu_request(resource_requirements["cpu"])
-        if "cpu_limit" in resource_requirements:
-            container_op.container.set_cpu_limit(resource_requirements["cpu_limit"])
-        if "gpu" in resource_requirements:
+        pod_spec: Dict[str, str] = kfp_component.pod_spec
+        if "memory" in pod_spec:
```

Comment:
If we're going to call this a `pod_spec`, can we align with the full structure of the `PodSpec` object please?

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#podspec-v1-core

Having an alternative schema structure to me will be super confusing to be honest. In this case I'd expect `memory` to be at `pod_spec['containers'][0]['resources']['requests']['mem']` in an ideal sense.

An alternative is not to name the variable `pod_spec` (likely the best course) so as not to confuse the two data structures, something like `container_attrs` would likely work.

Reply:
Thanks for the info. I was not aware of pod spec object and just came up with this name.